server/src/main/java/org/opensearch/index/mapper/IpFieldMapper.java

```diff
@@ -222,25 +224,42 @@ protected Object parseSourceValue(Object value) {
 
         @Override
         public Query termQuery(Object value, @Nullable QueryShardContext context) {
-            failIfNotIndexed();
+            failIfNotIndexedAndNoDocValues();
+            Query query;
             if (value instanceof InetAddress) {
-                return InetAddressPoint.newExactQuery(name(), (InetAddress) value);
+                query = InetAddressPoint.newExactQuery(name(), (InetAddress) value);
             } else {
                 if (value instanceof BytesRef) {
                     value = ((BytesRef) value).utf8ToString();
                 }
                 String term = value.toString();
+                if (term.contains("/")) {
+                    final Tuple<InetAddress, Integer> cidr = InetAddresses.parseCidr(term);
+                    query = InetAddressPoint.newPrefixQuery(name(), cidr.v1(), cidr.v2());
+                } else {
+                    InetAddress address = InetAddresses.forString(term);
+                    query = InetAddressPoint.newExactQuery(name(), address);
+                }
+            }
+            if (isSearchable() && hasDocValues()) {
+                String term = value.toString();
+                if (value instanceof String)
+                    value =  new BytesRef(((String) value).getBytes());
                 if (term.contains("/")) {
                     final Tuple<InetAddress, Integer> cidr = InetAddresses.parseCidr(term);
                     return InetAddressPoint.newPrefixQuery(name(), cidr.v1(), cidr.v2());
                 }
-                InetAddress address = InetAddresses.forString(term);
-                return InetAddressPoint.newExactQuery(name(), address);
+                return new IndexOrDocValuesQuery(query, SortedSetDocValuesField.newSlowExactQuery(name(), (BytesRef)(value)));
             }
+            if (hasDocValues()) {
+                return SortedSetDocValuesField.newSlowExactQuery(name(), ((BytesRef) value));
+            }
```

Comment:
If you get here, then you haven't passed through the logic above that converts a String value to `BytesRef`, so you will probably get a `ClassCastException`.